internal/schema-manager/mysql/mysql.go

```diff
@@ -127,10 +143,17 @@ func getDatabaseDataForSchemaDiff(
 			if err != nil {
 				return fmt.Errorf("failed to retrieve  database table constraints for schema %s: %w", schema, err)
 			}
-			for key, tableconstraint := range tableconstraints {
-				mu.Lock()
-				constraints[key] = tableconstraint
-				mu.Unlock()
+			for _, tableconstraint := range tableconstraints {
+				for _, nonFkConstraint := range tableconstraint.NonForeignKeyConstraints {
+					mu.Lock()
```

Comment:
same Q's here. idk if the mutex is needed unless more than one thing in a different goroutine is writing to this map.
If it _is_ needed, it's excessive to lock/unlock within this for loop I think. especially given you are just overwriting the map contents and not checking if there is a collision.